### PR TITLE
Query balances from bitmex

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/exchanges/bitmex.py [hunk 1]
```diff
@@ -52,32 +54,38 @@
...
-    return Asset(symbol).resolve_to_asset_with_oracles()
+    return symbol_to_asset_or_token(GlobalDBHandler.get_assetid_from_exchange_name(
+        exchange=Location.BITMEX,
+        symbol=(symbol_upper := symbol.upper()),
+        default=symbol_upper,
+    ))
 
 
-def trade_from_bitmex(bitmex_trade: dict) -> MarginPosition:
+def margin_trade_from_bitmex(bitmex_trade: dict) -> MarginPosition:
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:75]: Should you not also change the docstring here?

File: rotkehlchen/exchanges/bitmex.py [hunk 2]
```diff
@@ -129,11 +137,30 @@ def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> boo
...
+            return
+        try:
+            self.get_scales()
+        except RemoteError as e:
+            log.error(f'Failed to obtain bitmex decimals due to {e}. First connection not made')
+            return
+
         self.first_connection_made = True
 
+    def get_scales(self) -> None:
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:150]: Scales? Can you add a docstring?

File: rotkehlchen/exchanges/bitmex.py [hunk 3]
```diff
@@ -129,11 +137,30 @@ def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> boo
         return changed
 
     def first_connection(self) -> None:
+        if self.first_connection_made is True:
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:140]:
```suggestion
        if self.first_connection_made:
```

File: rotkehlchen/exchanges/bitmex.py [hunk 4]
```diff
@@ -129,11 +137,30 @@ def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> boo
...
+        except RemoteError as e:
+            log.error(f'Failed to obtain bitmex decimals due to {e}. First connection not made')
+            return
+
         self.first_connection_made = True
 
+    def get_scales(self) -> None:
+        mappings = {}
+        data = self._api_query(path='wallet/assets')
+        for entry in data:
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:153]: Use := operator here

File: rotkehlchen/exchanges/bitmex.py [hunk 5]
```diff
@@ -129,11 +137,30 @@ def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> boo
...
+            try:
+                mappings[entry['currency']] = entry['scale']
+            except KeyError as e:
+                log.error(f'Missing key {e} when querying decimals in {entry}')
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:157]: Shouldn't you quit with error here?

File: rotkehlchen/exchanges/bitmex.py [hunk 6]
```diff
@@ -212,67 +235,49 @@ def _api_query(
...
-        except DeserializationError as e:
-            msg = f'Bitmex API request failed. Failed to deserialized amount due to {e!s}'
-            log.error(msg)
-            return None, msg
+        for balance in resp:
+            try:
+                asset = bitmex_to_world(currency := balance['currency'])
+                raw_amount = balance['amount']
+            except (KeyError, UnknownAsset) as e:
+                log.error(e)
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:261]: You should have specific error message for key error

File: rotkehlchen/exchanges/bitmex.py [hunk 7]
```diff
@@ -212,67 +235,49 @@ def _api_query(
...
-            usd_value=usd_value,
-        )
-        log.debug(
-            'Bitmex balance query result',
-            currency='BTC',
-            amount=amount,
-            usd_value=usd_value,
-        )
+            if (decimals := self.asset_to_decimals.get(currency)) is None:
+                log.error(f'Unknown decimals for asset balance {balance} in bitmex. Skipping')
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:265]: Add asset name / symbol in this message

File: rotkehlchen/exchanges/bitmex.py [hunk 8]
```diff
@@ -283,22 +288,28 @@ def query_online_margin_history(
...
+                tx['transactType'] != 'RealisedPNL' or
+                (timestamp and not start_ts <= timestamp <= end_ts)
+            ):
                 continue
-            margin_trades.append(trade_from_bitmex(tx))
+
+            try:
+                margin_trades.append(margin_trade_from_bitmex(tx))
+            except (KeyError, UnknownAsset, DeserializationError):
+                log.error(f'Failed to process margin trade from bitmex {tx}. Skipping ...')
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/bitmex.py:312]: Same here. Special handling for key error needed

**Review by LefterisJP — COMMENTED** [review]
Some small comments

File: rotkehlchen/exchanges/bitmex.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/exchanges/bitmex.py:150]: It is what we call decimals for tokens/assets. Updated in the docstring

File: rotkehlchen/exchanges/bitmex.py (hunk 7 above)
Comment (yabirgb) [reply rotkehlchen/exchanges/bitmex.py:265]: it is included in balance. Balance is a mapping that contains all the info

**Review by LefterisJP — APPROVED** [review]
Lgtm

**Pull request merged**